### DISCUSSION TITLE
Sonoff RF  - Tasmota integration

### DISCUSSION THREAD

**Discussion by bgolab** [arendst/Tasmota#14413] [Q&A] [answered]
Hi,
I am looking for your insights/ tips regarding the Sonoff RF module (the one having the syn470r + EFM8BB10F4G) connected the main chip (ESP8266).

I just ordered this device and not sure how the RF board is connected to the ESP8266 (uart, spi, i2c?). Will do some reverse engineering once I get it.

So I am not sure if the Tasmota software is able to read the RF codes coming from the 4 buttons remote controller.

I simply would like to have more features than toggle the switch - I am talking about 4 buttons remote.

I'd be grateful if you could share some tips on this.

**Comment by barbudor** [comment, marked as answer]
ESP8266 Is connect through serial to the EFM8BB which is driving the RF part
All RF protocols are handled by the EFM MCU which provide data through serial.
If you are speaking about a Sonoff branded 4 button remote controller, it is likely to work
With any other remote, then it's lottery depending if the SOnoff firmware in the EFM chip can support it or not
The alternative Portish firmware can be uploaded into the EFM chip to extend this and can - at some extend - help supporting new protocols (see docs)
If you just ordered a Sonoff RF bridge, beware that they changed the design and since november some people have started to receive the new model with another mcu instead of the EFM. This model is not natively supported by Tasmota.
There is a proposed hack to shortcut the MCU and connect the RF directly to the ESP and use the RC-Switch library instead.

Everything in the docs or in other discussions topics.

**Comment by bgolab** [comment]
Thank you!
Just ordered pure sonoff RF switch (not the bridge version). I hope your valuable comments are applicable to sonoff RF device (probably the oldest version)

**Comment by barbudor** [comment]
AFAIK, not owning a RF switch but the old rf bridge, on the switch the RF part is not accessible to the ESP. It is limited to switch the relay on/off.

May need some advices from someone actually owning such device.

**Reply by bgolab** [reply]
I assume in the worst case scenario the direct connection between the RF chip and the ESP8266 (I mean bypassing the RF dedicated MCU) and will use the the RC-Switch library instead (as per your suggestion).

Thank you.
